### PR TITLE
chore: adding observability stack example

### PR DESCRIPTION
**Pull request by danielbdias**
This PR adds an example that shows the user how to create an observability stack, plug an API into it, and do some trace based tests.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: examples/observability-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Observability Stack example
```
Comment (jfermi) [inline examples/observability-stack/README.md:1]:
```suggestion
# Observability Stack Example
```

File: examples/observability-stack/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# run our Observability stack 
+docker compose up -d
+
+# install dependencies and run API
+npm install
+npm run with-telemetry
+
+```
+
+Now if you want to run it with Trace-based Tests, you can execute:
```
Comment (jfermi) [inline examples/observability-stack/README.md:17]:
```suggestion
If you want to run it with Trace-based Tests, you can execute:
```

**Review by jfermi — COMMENTED** [review]
Editing suggestions.

File: examples/observability-stack/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# install dependencies and run API
+npm install
+npm run with-telemetry
+
+```
+
+If you want to run it with Trace-based Tests, you can execute:
+
+```sh
+# run our Observability stack
```
Comment (adnanrahic) [inline examples/observability-stack/README.md:20]:
```suggestion
# run the observability stack 
```

**Review by adnanrahic — COMMENTED** [review]
Tiny suggestions.

File: examples/observability-stack/docker-compose.tracetest.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+services:
+  tracetest:
+    image: kubeshop/tracetest-agent:latest
+    platform: linux/amd64
+    depends_on:
+      otel-collector:
+        condition: service_started
+    environment:
+      TRACETEST_API_KEY: ${TRACETEST_API_KEY}
```
Comment (mathnogueira) [inline examples/observability-stack/docker-compose.tracetest.yaml:11]: We the agent will access `http://host.docker.internal:8080/`, it's better to add the extra_hosts parameter to make sure Linux users will be able to run this example:

```yaml
extra_hosts:
      - "host.docker.internal:host-gateway"
```

**Review by mathnogueira — APPROVED** [review]

**Pull request merged**